Library/Source/ShaderCompilerOpenGL.cpp

```diff
@@ -97,6 +104,28 @@ namespace Babylon
                 }
             }
 
+            auto combined = compiler->get_combined_image_samplers();
+            for (auto resource : resources.separate_samplers)
+            {
+                auto& samplerName = resource.name;
+                auto id = resource.id;
+                for (auto combine : combined)
+                {
+                    if (combine.sampler_id == id)
+                    {
+                        id = combine.combined_id;
+                        break;
+                    }
+                }
+                auto& fbn = compiler->get_fallback_name(id);
+                size_t pos = compiled.find(fbn);
+                while (pos != std::string::npos)
+                {
+                    compiled.replace(pos, fbn.size(), samplerName);
```

Comment:
We are doing more and more string replacing here. I wonder if there are better ways to deal with these differences.